pkg/crc/preflight/preflight_daemon_task_check_windows.go

```diff
@@ -217,7 +217,22 @@ func getDaemonPoshScriptContent() []byte {
 func checkDaemonPoshScript() error {
 	if exists := crcos.FileExists(daemonPoshScriptPath); exists {
 		// check the script contains the path to the current executable
-		if err := crcos.FileContentMatches(daemonPoshScriptPath, getDaemonPoshScriptContent()); err == nil {
+		content, err := os.ReadFile(filepath.Clean(daemonPoshScriptPath))
+		if err != nil {
+			return fmt.Errorf("Error opening file: %s: %s", daemonPoshScriptPath, err.Error())
+		}
+
+		binPath, err := os.Executable()
+		if err != nil {
+			return fmt.Errorf("Unable to find current crc executable path: %w", err)
+		}
+
+		drive := filepath.VolumeName(binPath)
+		binPathWithSmallCaseDriveLetter := strings.Replace(binPath, drive, strings.ToLower(drive), 1)
+		binPathWithUpperCaseDriveLetter := strings.Replace(binPath, drive, strings.ToUpper(drive), 1)
```

Comment:
Windows paths are case insensitive, I don't think we want to add more variations if there are other concrete situations where the case is not what we expect? Maybe we can force `binPath` to lower case in `getDaemonPoshScriptContent` to have a single canonical form for the binary path, and to make it independant from how the user start crc?

Reply:
> Do we get the correct behaviour with Crc setup followed by crc setup or crc.EXE setup followed by crc.exe setup?

these variants are producing correct behaviour, it doesn't effect the path, always normalized to `crc.exe`

> Maybe we can force binPath to lower case in getDaemonPoshScriptContent to have a single canonical form for the binary path, and to make it independant from how the user start crc?

yes, this will also work, and with much smaller changes, basically we have to match what is the constant here: https://github.com/crc-org/crc-extension/blob/571f6dbbb0cc95c4214c6ea35e32cb9af112045f/src/crc-cli.ts#L26

Reply:
is this only the drive letter which is problematic then? or can we also have issues with the dir name? `C:\program files\` vs `C:\Program Files`? I'm not clear on which go API is involved in the bug/what gets normalized/not normalized with go apis :)

A slightly hacky (or not ?) alternative would be to use `daemonPoshScriptTemplate` to create a regexp which we match against the installed template. With capturing regexp, we can even get the binary path used in the installed template to be able to run additional checks.

Reply:
> is this only the drive letter which is problematic then? or can we also have issues with the dir name? C:\program files\ vs C:\Program Files? I'm not clear on which go API is involved in the bug/what gets normalized/not normalized with go apis :)

yeah, only the drive letter is problematic, the go API is `os.Executable()` it is returning the path with smaller case drive letter when crc is invoked from the extension, this is because the invoked process's `PATH` env is modified to contain the path with _smaller_ case drive letter in the first entry of `PATH`

> A slightly hacky (or not ?) alternative would be to use daemonPoshScriptTemplate to create a regexp which we match against the installed template. With capturing regexp, we can even get the binary path used in the installed template to be able to run additional checks.

right, this will also be future proof, (what i am aiming is to make it behave the same irrespective of whether it is returning a lower case or upper case drive letter)

Reply:
> the dir names if different will also be an issue, but currently only small drive letter is causing the issue

I was indeed asking a general question, I understand that just takin care of `C:` VS `c:` will fix the extension bug. Though it seems command line users can easily hit this too.